pkg/jobs/registry.go

```diff
@@ -1111,32 +1119,37 @@ WHERE status IN ($1, $2, $3, $4, $5) ORDER BY created DESC`
 		_, runningOnNode := r.mu.jobs[*id]
 		r.mu.Unlock()
 
-		if notLeaseHolder := payload.Lease.NodeID != r.nodeID.DeprecatedNodeID(
-			multiTenancyIssueNo,
-		); notLeaseHolder {
-			// Another node holds the lease on the job, see if we should steal it.
-			if runningOnNode {
-				// If we are currently running a job that another node has the lease on,
-				// stop running it.
-				log.Warningf(ctx, "job %d: node %d owns lease; canceling", *id, payload.Lease.NodeID)
-				r.unregister(*id)
-				continue
-			}
-			nodeStatus, ok := nodeStatusMap[payload.Lease.NodeID]
-			if !ok {
-				// This case should never happen.
-				log.ReportOrPanic(ctx, nil, "job %d: skipping: no liveness record for the job's node %d",
-					log.Safe(*id), payload.Lease.NodeID)
-				continue
-			}
-			if nodeStatus.isLive {
-				if log.V(2) {
-					log.Infof(ctx, "job %d: skipping: another node is live and holds the lease", *id)
+		// If we're running as a tenant (!ok), then we are the sole SQL server in
+		// charge of its jobs and ought to adopt all of them. Otherwise, look more
+		// closely at who is running the job and whether to adopt.
+		if nodeID, ok := r.nodeID.OptionalNodeID(); ok {
+			if notLeaseHolder := payload.Lease.NodeID != nodeID; notLeaseHolder {
+				// Another node holds the lease on the job, see if we should steal it.
+				if runningOnNode {
+					// If we are currently running a job that another node has the lease on,
+					// stop running it.
+					log.Warningf(ctx, "job %d: node %d owns lease; canceling", *id, payload.Lease.NodeID)
+					r.unregister(*id)
+					continue
+				}
+				nodeStatus, ok := nodeStatusMap[payload.Lease.NodeID]
+				if !ok {
+					// This case should never happen.
+					log.ReportOrPanic(ctx, nil, "job %d: skipping: no liveness record for the job's node %d",
+						log.Safe(*id), payload.Lease.NodeID)
+					continue
+				}
+				if nodeStatus.isLive {
+					if log.V(2) {
+						log.Infof(ctx, "job %d: skipping: another node is live and holds the lease", *id)
+					}
+					continue
 				}
-				continue
 			}
 		}
-		// Below we know that this node holds the lease on the job.
+
+		// Below we know that this node holds the lease on the job, or that we want
```

Comment:
It's fine that we hit this with `runningOnNode == true`, right? It will hit `currentlyRunning` below and bail out, and also this is the same for single-tenant jobs.